### PR TITLE
mgr/rbd_support: create mirror snapshots asynchronously

### PR DESCRIPTION
**Pull request by trociny**
To scale up with number of images.

Fixes: https://tracker.ceph.com/issues/47827
Signed-off-by: Mykola Golub <mgolub@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by trociny** [comment]
@dillaman thanks. updated.

**Review by dillaman — APPROVED** [review]
lgtm

**Comment by dillaman** [comment]
Looks like the shaman builders failed:

```
/home/jenkins-build/build/workspace/ceph-dev-new-build/ARCH/x86_64/AVAILABLE_ARCH/x86_64/AVAILABLE_DIST/centos8/DIST/centos8/MACHINE_SIZE/gigantic/release/16.0.0-6724-gc91ead17/rpm/el8/BUILD/ceph-16.0.0-6724-gc91ead17/src/librbd/api/Mirror.cc:2047:53: error: class template argument deduction failed:
                                            on_finish);
```

File: src/librbd/api/Mirror.cc [hunk 1]
```diff
@@ -1978,43 +2015,48 @@ int Mirror<I>::image_info_list(
...
-  cls::rbd::MirrorImage mirror_image;
-  r = cls_client::mirror_image_get(&ictx->md_ctx, ictx->id,
-                                   &mirror_image);
-  if (r == -ENOENT) {
-    return -EINVAL;
-  } else if (r < 0) {
-    lderr(cct) << "failed to retrieve mirror image" << dendl;
-    return r;
-  }
+      auto ctx = new C_ImageSnapshotCreate(ictx, snap_create_flags, snap_id,
```
Comment (dillaman) [inline src/librbd/api/Mirror.cc:2046]:
```suggestion
      auto ctx = new C_ImageSnapshotCreate<I>(ictx, snap_create_flags, snap_id,
```

Comment (trociny) [reply src/librbd/api/Mirror.cc]: @dillaman Thanks! Updated.

**Comment by dillaman** [comment]
Looks like there are python test failures [1]

[1] http://qa-proxy.ceph.com/teuthology/jdillaman-2020-10-29_12:26:35-rbd-wip-jd-testing-distro-basic-smithi/5572253/teuthology.log

**Comment by trociny** [comment]
> Looks like there are python test failures [1]
> 
> [1] http://qa-proxy.ceph.com/teuthology/jdillaman-2020-10-29_12:26:35-rbd-wip-jd-testing-distro-basic-smithi/5572253/teuthology.log

Forgot to remove the image in `test_aio_open`. Sorry for this. It should be fixed now.

**Pull request merged**